lib/ecto/changeset.ex

```diff
@@ -1217,6 +1217,19 @@ defmodule Ecto.Changeset do
     raise ArgumentError, "field names given to change/put_change must be atoms, got: `#{inspect(key)}`"
   end
 
+  defp put_change(data, changes, errors, valid?, key, new_value, {:parameterized, type, params}) do
+    old_value = Map.get(data, key)
+
+    if type.equal?(old_value, new_value, params) do
```

Comment:
My plan is to not have `equal?` for parameterized types. Instead, they can simply return `:ignore` or similar. That's because sometimes checking if they are the same or not is non-straight-forward. 

For the rest, I will only know if this is good once we make it work with Ecto.Embedded. :)

Reply:
This is resolved, but removing callback is not part of this PR :-)

Reply:
The callback will still stay, so no worries! :)